perf/mongodb.go

```diff
@@ -1,9 +1,127 @@
 package perf
 
 import (
-	mongowrapper "github.com/opencensus-integrations/gomongowrapper"
+	"context"
+	"fmt"
+	"strings"
+	"sync"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/event"
+	"go.opencensus.io/trace"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
+
+	"go.viam.com/utils"
 )
 
 func registerMongoDBViews() error {
-	return mongowrapper.RegisterAllViews()
+	// TODO(erd): add views
+	return nil
+}
+
+// from https://github.com/entropyx/mongo-opencensus
+
+type config struct {
+	sampler trace.Sampler
+}
+
+// MongoDBMonitorOption represents an option that can be passed to NewMongoDBMonitor.
+type MongoDBMonitorOption func(*config)
+
+// WithMongoDBMonitorSampler set a sampler for all started spans.
+func WithMongoDBMonitorSampler(sampler trace.Sampler) MongoDBMonitorOption {
+	return func(cfg *config) {
+		cfg.sampler = sampler
+	}
+}
+
+type spanKey struct {
+	ConnectionID string
+	RequestID    int64
+}
+
+type monitor struct {
+	sync.Mutex
+	spans map[spanKey]*trace.Span
+	cfg   *config
+}
+
+func (m *monitor) Started(ctx context.Context, evt *event.CommandStartedEvent) {
+	connString := connectionString(evt)
+	b, err := bson.MarshalExtJSON(evt.Command, false, false)
+	if err != nil {
+		utils.UncheckedError(err)
+		return
+	}
+	_, span := trace.StartSpan(ctx, evt.CommandName, trace.WithSampler(m.cfg.sampler))
+	span.AddAttributes(trace.StringAttribute(ext.DBInstance, evt.DatabaseName),
+		trace.StringAttribute("db.system", "mongodb"),
+		trace.StringAttribute("db.operation", string(b)),
+		trace.StringAttribute("db.connection_string", connString),
```

Comment:
Does this accidentally leak secrets when password is used in connection string? Maybe just use name of the cluster/host?

Reply:
no, see the function it derives it from